### DISCUSSION TITLE
Real-time transcription with Whisper on a desktop app

### DISCUSSION THREAD

**Discussion by chidiwilliams** [openai/whisper#246] [Show and tell]
For many who would rather use a desktop app than a CLI, I made a cross-platform app called Buzz.

![Buzz screenshot](https://github.com/chidiwilliams/buzz/blob/1cf2e9769899a80ff08bd3e6e4b0f541f919acdc/assets/buzz.png?raw=true)

Here's a demo: https://www.loom.com/share/564b753eb4d44b55b985b8abd26b55f7

Repo: https://github.com/chidiwilliams/buzz

**Comment by DeysonOrtiz** [comment]
Great work! 🙂

**Comment by MbuguaDavid** [comment]
Thank you for the cross-platform app. Highly appreciated.

Did a video to let folks know about it.
https://youtu.be/bWayH2AvFDU

Thanks for this.

**Comment by Abdurrafey-Siddiqui** [comment]
> For many who would rather use a desktop app than a CLI, I made a cross-platform app called Buzz.
> 
> ![Buzz screenshot](https://github.com/chidiwilliams/buzz/blob/1cf2e9769899a80ff08bd3e6e4b0f541f919acdc/assets/buzz.png?raw=true)
> 
> Here's a demo: https://www.loom.com/share/564b753eb4d44b55b985b8abd26b55f7
> 
> Repo: https://github.com/chidiwilliams/buzz

Hi, is there a way fos whisper to recognise more words within this app. Like we can manually add words so that whisper doesn't get it wrong.

**Reply by snortlax** [reply]
This is interesting, I'm not not a programmer (aspiring to be one) so I have no idea how to look into your code and see how it functions so I thought i'd just ask. Sorry if the answer is painfully obvious.

I am assuming you are recording the audio in chunks of the delay size and run it through the model right? If so, how do you deal with words that are cut off by the edge of the chunks?

**Reply by chidiwilliams** [reply]
Yes, that's correct. You're right that words cut off by the chunks will sometimes get wrongly transcribed. The app also has a way to import audio files, which would give more accurate transcriptions than the live recording.

**Comment by forest428-gmail** [comment]
Very nice, which IDE did you use to develop the project?

**Reply by chidiwilliams** [reply]
VS Code.

**Comment by wwbnjsace** [comment]
> For many who would rather use a desktop app than a CLI, I made a cross-platform app called Buzz.
> 
> ![Buzz screenshot](https://github.com/chidiwilliams/buzz/blob/1cf2e9769899a80ff08bd3e6e4b0f541f919acdc/assets/buzz.png?raw=true)
> 
> Here's a demo: https://www.loom.com/share/564b753eb4d44b55b985b8abd26b55f7
> 
> Repo: https://github.com/chidiwilliams/buzz

this can not do streaming asr .do you know how to do streaming asr?any code?

**Reply by chidiwilliams** [reply]
What do you mean by streaming ASR?

**Comment by nyadla-sys** [comment]
Please see my project below, which uses the Whisper Tiny Tflite Model to implement audio streaming..
[Streaming using TFLite model](https://github.com/usefulsensors/openai-whisper/tree/main/stream)

**Reply by chidiwilliams** [reply]
Looks good. Have you seen [Whisper.cpp](https://github.com/ggerganov/whisper.cpp)? You might find the project interesting also.

**Comment by vbalashi** [comment]
Can you tell me why does it take 20 minutes to complete this via CLI for a 2-minute MP3 file on my MacPro 2020 with 16GB memory:
whisper p1l21.mp3 --language Dutch --model large

while Buzz finishes the same thing in just 2 minutes?

**Reply by chidiwilliams** [reply]
Hey. This should be because you're selecting the large model ("--model large") when running from the CLI. Depending on what quality you select while running Buzz, you may be using one of the smaller models for the transcription, which run faster but with slightly lower accuracy.

**Reply by vbalashi** [reply]
in Buzz I also use large model. There was an idea that "beam size" parameter makes sense. Just need to discover what is used in Buzz

**Comment by WanZunGe** [comment]
![image](https://user-images.githubusercontent.com/100510668/206483267-e6b01df2-696f-426c-9661-e3ba6e299f6e.png)
Why do I get errors every time I run the program

**Reply by chidiwilliams** [reply]
What version of the app are you using?

**Reply by WanZunGe** [reply]
![image](https://user-images.githubusercontent.com/100510668/206516348-a2eb962f-0505-4598-aab8-2f47035aeb3f.png)
latest version

**Reply by WanZunGe** [reply]
The version of windows I was using, the normal installation was successful, and then the error was displayed as soon as I ran it. At first I thought it was because I did not install it on the C drive, but later I uninstalled it and installed it on the C drive, and the result was the same

**Reply by chidiwilliams** [reply]
Could you try this version? https://github.com/chidiwilliams/buzz/releases/tag/v0.5.8. I think I broke the v0.6 and above builds.

**Reply by WanZunGe** [reply]
Perfect solution. Thank you. This is a great app

**Comment by jamalissa** [comment]
Is it possible to record the pc sounds instead of the microphone "just like screen-recording apps" so it can record from a Zoom meeting or a locally saved video?

**Comment by Gldkslfmsd** [comment]
Hi, I recommend integrating [Whisper-Streaming](https://github.com/ufal/whisper_streaming) into this tool, for self-adaptive latency based on the actual source complexity.
https://github.com/openai/whisper/discussions/1978

**Comment by learndeeplearn** [comment]
best man ever :)

**Comment by MayMory** [comment]
I WANT THIS ON WIN7.....

**Comment by learndeeplearn** [comment]
I bought the mac app, could you please add some summary feature?)
